src/sentry/integrations/github/client.py

```diff
@@ -632,10 +633,14 @@ def get_labels(self, repo: str) -> Sequence[Any]:
         """
         return self.get(f"/repos/{repo}/labels", params={"per_page": 100})
 
-    def check_file(self, repo: Repository, path: str, version: str) -> BaseApiResponseX:
+    def check_file(
+        self, repo: Repository, path: str, version: str | None
+    ) -> BaseApiResponseX | None:
```

Comment:
It seems like on a cache miss it sets the cache for that key and returns result so this can't be None. I don't think you will get type failure is the child type is subset of the parent.

Reply:
you're right, thanks for noticing!